tests/main/interfaces-bluetooth-control/task.yaml

```diff
@@ -57,4 +57,4 @@ execute: |
 
     echo "And the snap is able to read dev"
     cat "$BTDEV"/*/power/control | tee control
-    [ "$(su -l -c '/snap/bin/generic-consumer.cmd cat '"$BTDEV"'/*/power/control' test)'" = "$(cat control)" ]
+    [ "$(su -l -c '/snap/bin/generic-consumer.cmd cat '"$BTDEV"'/*/power/control' test)" = "$(cat control)" ]
```

Comment:
Let's take this apart:

```
[ "$(su -l -c '/snap/bin/generic-consumer.cmd cat '"$BTDEV"'/*/power/control' test)" = "$(cat control)" ]
```
Dropping the outer test syntax we check if the first line evaluates to the second line:
```
"$(su -l -c '/snap/bin/generic-consumer.cmd cat '"$BTDEV"'/*/power/control' test)"
"$(cat control)"
```
Focusing on the first of those lines, dropping the outer double quotes, shell evaluate-and-expand syntax we get:
```
su -l -c '/snap/bin/generic-consumer.cmd cat '"$BTDEV"'/*/power/control' test
```

Which is running a command as the user `test` via `su`. Looking at the command now:
```
'/snap/bin/generic-consumer.cmd cat '"$BTDEV"'/*/power/control'
```
To me this was weird at first, the sequence of quoted string segments is:
- `'/snap/bin/generic-consumer.cmd cat '`
- `"$BTDEV"`
- `'/*/power/control'`

This raised a red flag in my mind until I played with shell expansion and noticed that there are indeed no spaces between the three quoted string segments. Thus for shell this is a single argument with variable expand and quote semantics along the way.

Shell is tricky but this looks correct. Thanks @sergiocazzolato